_posts/2021-11-19-migrating-kafka-with-mirror-maker2.md

```diff
@@ -0,0 +1,227 @@
+features needed to cover our requirements. The most important features are:
+
+* **Topic topology replication**, as we need to have the same structure of
+the topic and partitions in the target Kafka cluster.
+* **Consumer group offset synchronization**, as we need to consume the data
+from the target Kafka cluster at the same point the consumers stopped in
+the source Kafka cluster. This feature tries to avoid consuming duplicated
+messages in the target cluster.
+* **Dynamic detection of changes** in source Kafka cluster and replicate in
+target Kafka cluster. Every change must be replicated in the target Kafka cluster.
+* **Replication ACL** rules that manage access to source brokers and topics
+apply to target topics. If we are not using the [User Operator of
+Strimzi](https://strimzi.io/docs/operators/latest/using.html#assembly-using-the-user-operator-str),
+in that case we should recreate the KafkaUsers in the target cluster.
+More info [here](https://strimzi.io/docs/operators/latest/using.html#con-mirrormaker-acls-str).
+
+The synchronization processes are managed by a [set of properties](https://kafka.apache.org/documentation/#georeplication-flow-configure)
+to set up intervals and checkpoints. It is important to review and
+check these properties and the applications movements to avoid or reduce
+the duplication process of some data.
+
+MirrorMaker2 [could rename the topics](https://kafka.apache.org/documentation/#georeplication-topic-naming)
+in the target Kafka cluster (remote topics) with some patterns
+and conventions. During a migration process this kind of changes are
+not a good idea, it is a best practice to maintain a 1-1 relationship
+between the source and kafka cluster. 
+
+### MirrorMaker2 Connectors
+
+MirrorMaker2 deploys a KafkaConnect cluster with a set of different
+Kafka Connectors to execute the migration and synchronization between clusters.
+
+* **MirrorSourceConnector** is responsible for topics, data replication based
+on a replication policy, and other synchronization tasks such as ACL
+rules, renaming topics, …
+* **MirrorCheckpointConnector** is responsible for track offset for
+consumer groups and synchronizing into the target Kafka cluster.
+* **MirrorHeartbeatConnector** is responsible for checking the connectivity
+between clusters using a set of heartbeats and checks.
+
+The action combinated of all of them, using a set of internal topics,
+executes this complex process of migrating and synchronizing both
+Kafka clusters.
+
+## Red Hat AMQ Streams Operators make migrations happier
+
+Operating an Apache Kafka system with all these components in an
+OpenShift environment sounds like a huge and complex activity, however
+the reality is that is very easy thanks to the powerful
+[Kubernetes Operators](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/)
+provided by the great community of [Strimzi](https://strimzi.io/).
+
+Red Hat AMQ Streams provides a full list of operators to help streamline
```

Comment:
Again, using Strimzi would be probably the proper way here.

Reply:
Idem as above! Fixing!